File: db4o.net/Db4objects.Db4o.CS/Db4objects.Db4o.CS/CS/Internal/Messages/MReadReaderById.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4objects.Db4o.CS.Internal.Messages;
using Db4objects.Db4o.Ext;
using Db4objects.Db4o.Internal;

namespace Db4objects.Db4o.CS.Internal.Messages
{
	public class MReadReaderById : MsgD, IMessageWithResponse
	{
		public Msg ReplyFromServer()
		{
			ByteArrayBuffer bytes = null;
			// readWriterByID may fail in certain cases, for instance if
			// and object was deleted by another client
			try
			{
				lock (ContainerLock())
				{
					bytes = Container().ReadBufferById(Transaction(), _payLoad.ReadInt(), _payLoad.ReadInt
						() == 1);
				}
				if (bytes == null)
				{
					bytes = new ByteArrayBuffer(0);
				}
			}
			catch (Db4oRecoverableException exc)
			{
				throw;
			}
			catch (Exception exc)
			{
				throw new Db4oRecoverableException(exc);
			}
			return Msg.ReadBytes.GetWriter(Transaction(), bytes);
		}
	}
}

File: db4o.cs/src/com/db4o/cs/internal/messages/MReadReaderById.java
package com.db4o.cs.internal.messages;

import com.db4o.ext.*;
import com.db4o.internal.*;

public class MReadReaderById extends MsgD implements MessageWithResponse {
	
	public final Msg replyFromServer() {
		ByteArrayBuffer bytes = null;
		// readWriterByID may fail in certain cases, for instance if
		// and object was deleted by another client
		try {
			synchronized (containerLock()) {
				bytes = container().readBufferById(transaction(), _payLoad.readInt(), _payLoad.readInt()==1);
			}
			if (bytes == null) {
				bytes = new ByteArrayBuffer(0);
			}
		}
		catch(Db4oRecoverableException exc) {
			throw exc;
		}
		catch(Throwable exc) {
			throw new Db4oRecoverableException(exc);
		}
		return Msg.READ_BYTES.getWriter(transaction(), bytes);
	}
}